src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -16,7 +17,7 @@
   protected static final HostAndPort DEFAULT_HOST_AND_PORT = new HostAndPort(Protocol.DEFAULT_HOST,
       Protocol.DEFAULT_PORT);
 
-  private HostAndPort hostAndPort = DEFAULT_HOST_AND_PORT;
+  private final AtomicReference<HostAndPort> hostAndPort = new AtomicReference<>(DEFAULT_HOST_AND_PORT);
```

Comment:
Do you really need AtomicReference or a `volatile` will do?

Reply:
volatile will work fine, we are not doing any CAS operations so it should be the same